### PR TITLE
Bug 1874804 - Replace ClearFlags with Engine.BrowsingData

### PR DESCRIPTION
**Pull request by jackyzy823**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1874804

### CODE REVIEW

**Review by Mukuljangir372 — APPROVED** [review]

**Comment by gabrielluong** [comment]
@Amejia481 I think you will know GeckoSitePermissionsStorage.kt best, would you have time to review this or find someone else to do so?

**Comment by Amejia481** [comment]
> @Amejia481 I think you will know GeckoSitePermissionsStorage.kt best, would you have time to review this or find someone else to do so?

Sure!

File: android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt [hunk 1]
```diff
@@ -340,13 +341,16 @@ class GeckoSitePermissionsStorage(
...
+                origin,
+                Engine.BrowsingData.select(PERMISSIONS).types.toLong(),
+            ).await()
         }
     }
 
     @VisibleForTesting
     internal fun clearAllPermissionsGeckoCache() {
-        geckoStorage.clearData(ClearFlags.PERMISSIONS)
+        geckoStorage.clearData(Engine.BrowsingData.select(PERMISSIONS).types.toLong())
```
Comment (Amejia481) [inline android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:353]: @jackyzy823 I think we could call directly `Engine.BrowsingData.PERMISSIONS`, as[ Engine.BrowsingData.PERMISSIONS](https://github.com/mozilla-mobile/firefox-android/blob/3b2bd69f8ebfdc4b74af71162636a75967ed6556/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt#L41) and [StorageController.PERMISSIONS](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/StorageController.java#51) are the same value (`64`). What do you think?

Comment (jackyzy823) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:353]: Yes, but i think we could keep the same way passing flags to clearData* methods.

Comment (Amejia481) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:353]: Could you clarify why do we need the extra step of using `Engine.BrowsingData.select`?

Comment (Amejia481) [reply android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt:353]: Could be use  `geckoStorage.clearData(Engine.BrowsingData.PERMISSIONS.toLong())` instead?

**Comment by gabrielluong** [comment]
@jackyzy823 @Amejia481 Hey folks, is this something we're still looking to pursue landing or could we close this PR?

**Comment by gabrielluong** [comment]
Hello @jackyzy823, Thank you so much for your contribution. We have moved our firefox-android repo over to [mozilla-central](https://hg.mozilla.org/mozilla-central). Please feel free to reopen a new PR against the new repo. You will have to clone the new repository and reapply your changes. Thank you again, and sorry for the inconvenience.

For more information, please see https://github.com/mozilla-mobile/firefox-android/wiki#upcoming-migration-to-mozilla-central.

**Pull request closed without merging**